test/sh_spec.py

```diff
@@ -760,29 +818,46 @@ def RunCases(cases, case_predicate, shells, env, out, opts):
             elif not opts.timeout_bin and actual['status'] == 124:
                 cell_result = Result.TIMEOUT
             else:
-                messages = []
-                cell_result = Result.PASS
-
-                # TODO: Warn about no assertions?  Well it will always test the error
-                # code.
-                assertions = CreateAssertions(case, sh_label)
-                for a in assertions:
-                    result, msg = a.Check(sh_label, actual)
-                    # The minimum one wins.
-                    # If any failed, then the result is FAIL.
-                    # If any are OK, but none are FAIL, the result is OK.
-                    cell_result = min(cell_result, result)
-                    if msg:
-                        messages.append(msg)
-
-                if cell_result != Result.PASS or opts.details:
+                best_result = 0
+                best_messages = None
+                best_qualifier_level = 0
+
+                for assertions in CreateAssertions(case, sh_label):
+                    messages = []
+                    cell_result = Result.PASS
+                    # TODO: Warn about no assertions?  Well it will always test the
+                    # error code.
+                    for a in assertions:
+                        result, msg = a.Check(sh_label, actual)
+                        # The minimum one wins.
+                        # If any failed, then the result is FAIL.
+                        # If any are OK, but none are FAIL, the result is OK.
+                        cell_result = min(cell_result, result)
+                        if msg:
+                            messages.append(msg)
+
+                    qualifier_level = max(
+                        [a.QualifierLevel() for a in assertions])
+
+                    # We choose the results of the assertion set with the
```

Comment:
Hm I would have to think about the "mildest failure" logic ... where else would we use that?

Reply:
I mean, if the test result (status/stdout/stderr) satisfies e.g. both the "OK" requirements and the "N-I" requirements, the final result we show in the table would be "OK". By "mild", I mean the ordering of `PASS > OK > N-I > BUG > FAIL`.